cfme/tests/intelligence/reports/test_canned_corresponds.py

```diff
@@ -7,15 +7,30 @@
 from utils.appliance.implementations.ui import navigate_to
 from utils.net import ip_address, resolve_hostname
 from utils.providers import get_crud_by_name
-from utils import version
+from utils.appliance import get_or_create_current_appliance
+from utils import version, testgen
 from cfme import test_requirements
 
 provider_props = partial(details_page.infoblock.text, "Properties")
 
+pytest_generate_tests = testgen.generate(classes=[InfraProvider], scope='module')
+
+pytestmark = [pytest.mark.usefixtures('setup_provider')]
+
+'''
+@pytest.fixture(scope="module")
```

Comment:
Check out the new provider fixtures, you can simplify your fixture here:
https://github.com/ManageIQ/integration_tests/blob/master/fixtures/provider.py#L29